main/tests/cypress/cypress/integration/project/grid/all-column/flag.spec.js

```diff
@@ -0,0 +1,34 @@
+describe(__filename, function () {
+  it('Ensure flag is visible and toggle on/off', function () {
+    cy.loadAndVisitProject('food.mini');
+    cy.get('.odd > :nth-child(2) > .data-table-flag-off')
+      .should('exist')
+      .click();
+    cy.get('#notification > div:nth-child(2) > span:nth-child(1)').should(
```

Comment:
@S-Harshit there's a command to check OR notifications
` cy.assertNotificationContainingText('Flag row 1');`

Reply:
@S-Harshit the way you've done the test is technically working, but the selectors you've used could be improved.
Using odd/even can make one think you are really testing the behavior of odd/even rows, and that's not the case here

What you can do is to select rows by their index, columns a well.
example
``` 
cy.get('.data-table tr:nth-child(1) td:nth-child(2) a').should('have.class', 'data-table-flag-off').click();
cy.get('.data-table tr:nth-child(1) td:nth-child(2) a').should('have.class', 'data-table-flag-on');
```
This way the selector is meaningful, in human language that would says: Give the first row, then give me the second column, then click

Reply:
I have changed the selectors and the command to check notifications as suggested